src/pilot/assign_role.py

```diff
@@ -1367,15 +1367,21 @@ def change_physical_disk_state_wait(
         mode, controllers_to_physical_disk_ids)
 
     job_ids = []
+    all_realtime_controllers = True
     if change_state_result['commit_required_ids']:
+        disk_controllers = {c.id: c for c in
+                            drac_client.list_raid_controllers()}
         for controller_id in change_state_result['commit_required_ids']:
+            all_realtime_controllers = all_realtime_controllers and \
+                disk_controllers[controller_id].supports_realtime
             job_id = drac_client.commit_pending_raid_changes(
-                controller_id, reboot=False, start_time=None)
+                controller_id, reboot=False, start_time=None,
+                realtime=disk_controllers[controller_id].supports_realtime)
             job_ids.append(job_id)
 
     result = True
     if job_ids:
-        if change_state_result['is_reboot_required']:
+        if all_realtime_controllers:
```

Comment:
I believe the above line should be:
if not all_realtime_controllers:
since we only want to reboot if there are 1 or more controllers that do not support realtime.

Reply:
Yes thanks chris for pointing out the issue, i had updated in my local repo and will update in remote repo.